devito/types/sparse.py

```diff
@@ -202,6 +202,12 @@ def __subfunc_setup__(self, suffix, keys, dtype=None, inkwargs=False, **kwargs):
             else:
                 dtype = dtype or self.dtype
 
+        # Complex coordinates are not valid, so fall back to corresponding
+        # real floating point type if dtype is complex.
+        if issubclass(dtype, np.complexfloating):
```

Comment:
Weird my comment disapearded. 

`dtype = self.dtype(0).real.__class__` is safer. For example, if someone uses fp16 sparse this would automatically upcast all subfunctions

Reply:
Oh that's much better, I was looking for something like that